react/shippingStateMachine/useShippingStateMachine.ts

```diff
@@ -75,8 +74,15 @@ const useShippingStateMachine = ({
           receiverName,
         })
       },
-      tryToCreateAddress: (_, event) => {
-        return insertAddress(event.address)
+      tryToCreateAddress: async (_, event) => {
+        const res = await insertAddress(event.address)
+        const { success, orderForm } = res;
+
+        if (success && orderForm?.shipping.selectedAddress) {
+          setAddress(orderForm.shipping.selectedAddress)
```

Comment:
do we need to add this call to other methods as well? such as `tryToSelectAddress`, etc. I know I had to add it to `updateSelectedAddress` but I wonder if we need to add it to the other ones

Reply:
I'll do some reviewing... I think I was working with a branch created from an old master. When I just tried to create this PR, I had to do a rebase and I've pulled your code from some days ago that do this. And I'm wondering exactly this too.

I'll review the code and rerun this and post any update here soon.

Reply:
nice! thanks